cern_access/indico_cern_access/tests/request_test.py

```diff
@@ -0,0 +1,93 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from datetime import date
+
+import pytest
+
+from indico.modules.events.registration.util import create_registration, make_registration_form
+
+from indico_cern_access.tests.conftest import PERSONAL_DATA
+from indico_cern_access.util import RegformDataMode
+
+
+@pytest.mark.parametrize('mock_access_request',
+                         [{
+                             'mode': RegformDataMode.during_registration,
+                             'personal_data': PERSONAL_DATA
+                         }],
+                         indirect=True)
+@pytest.mark.usefixtures('smtp', 'mock_access_request')
+def test_during_registration_positive(dummy_regform):
+    """Data can be provided during registration (it is)."""
+    assert dummy_regform.cern_access_request is not None
+
+    form = make_registration_form(dummy_regform)(csrf_enabled=False)
+    assert form.validate()
+
+    registration = create_registration(dummy_regform, form.data)
+    assert registration.cern_access_request is not None
+    assert registration.cern_access_request.nationality == 'PT'
+    assert registration.cern_access_request.birth_date == date(2000, 3, 2)
+    assert registration.cern_access_request.birth_place == 'Freixo de Espada as Costas'
+
+
+@pytest.mark.parametrize('mock_access_request',
+                         [{
+                             'mode': RegformDataMode.during_registration
+                         }],
+                         indirect=True)
+@pytest.mark.usefixtures('smtp', 'mock_access_request')
+def test_during_registration_negative(dummy_regform):
+    """Data can be provided during registration (it is not)."""
+    form = make_registration_form(dummy_regform)(csrf_enabled=False)
+    assert form.validate()
+
+    registration = create_registration(dummy_regform, form.data)
+    assert registration.cern_access_request is None
+
+
+@pytest.mark.parametrize('mock_access_request',
+                         [{
+                             'mode': RegformDataMode.during_registration_required
+                          }],
```

Comment:
too much indentation (probably easier to just have the whole `[{...}]` part in one line here)

Reply:
Why is it too much indentation?

Reply:
```python
@pytest.mark.parametrize('mock_access_request',
                         [{
                             'mode': RegformDataMode.during_registration_required,
                             'personal_data': PERSONAL_DATA
                          }],
                         indirect=True)
def foo():
    pass
```

vs

```python
@pytest.mark.parametrize('mock_access_request',
                         [{
                             'mode': RegformDataMode.during_registration_required,
                             'personal_data': PERSONAL_DATA
                         }],
                         indirect=True)
def foo():
    pass
```

Reply:
Ah, right. I hadn't noticed that. I wonder why flake8 doesn't complain...